tutorials/ner_pytorch_medical/scripts/preprocess.py

```diff
@@ -0,0 +1,175 @@
+from collections import defaultdict
+import random
+from typing import List
+import tarfile
+import shutil
+import typer
+import srsly
+from pathlib import Path
+import spacy
+from spacy.language import Language
+from spacy.tokens import Doc, DocBin, Span
+from spacy.util import get_words_and_spaces, filter_spans
+
+
+random.seed(42)
+
+
+def main(
+    input_dir: Path = typer.Argument(..., exists=True),
+    output_dir: Path = typer.Argument(...),
+    beth_train_tar_name: str = "i2b2_Beth_Train_Release.tar.gz",
+    partners_train_tar_name: str = "i2b2_Partners_Train_Release.tar.gz",
+    test_zip_name: str = "Task_1C.zip",
+    merge_docs: bool = True,
+):
+    """Extract and preprocess raw n2c2 2011 Challenge data into spaCy DocBin format.
+    input_dir (Path, optional): Input directory with raw downloads from Harvard DBMI Portal.
```

Comment:
We'd typically annotate this as 
```suggestion
    input_dir (Optional[Path]): Input directory with raw downloads from Harvard DBMI Portal.
```

Reply:
Right, I use this vscode extension to generate docstrings so I forget to fix this issue with optional params sometimes. Will do